### PR TITLE
statistics: add test for analyze single partition

### PR DESCRIPTION
**Pull request by rebelice**
### What problem does this PR solve?

Problem Summary:

Add a test to verify the two behaviors：

1. When analyzing a single partition, we will try to build global stats. If another partition doesn't have stats, be failed and return warnings.
2. After analyzing a single partition, we should have stats in `mysql.stats_fm_sketch`

### Check List 

Tests 

- Unit test

Side effects

### Release note 

- No release note

### CODE REVIEW

**Review by Reminiscent — COMMENTED** [review]
Please fix the failed test. Rest LGTM

File: statistics/handle/handle_test.go [hunk 1]
```diff
@@ -1419,6 +1419,25 @@ func (s *testStatsSuite) TestAnalyzeWithDynamicPartitionPruneMode(c *C) {
...
+	tk.MustExec("set @@tidb_analyze_version = 2")
+	tk.MustExec(`create table t (a int, key(a)) partition by range(a) 
+					(partition p0 values less than (10),
+					partition p1 values less than (22))`)
+	tk.MustExec(`insert into t values (1), (2), (3), (10), (11)`)
+	tk.MustQuery("select count(*) from mysql.stats_fm_sketch").Check(testkit.Rows("0"))
+	tk.MustExec("analyze table t partition p0 with 1 topn, 2 buckets")
+	tk.MustQuery("show warnings").Check(testkit.Rows(
+		"Warning 8131 Build global-level stats failed due to missing partition-level stats",
+		"Warning 8131 Build global-level stats failed due to missing partition-level stats",
```
Comment (qw4990) [inline statistics/handle/handle_test.go:1436]: How about adding the name of its index or column into it:
```
... build {index_name} global-level stats failed ...
... build {column_name} global-level stats failed ...
```

**Review by Reminiscent — APPROVED** [review]
/LGTM

File: statistics/handle/handle.go [hunk 2]
```diff
@@ -367,16 +367,28 @@ func (h *Handle) MergePartitionStats2GlobalStats(sc sessionctx.Context, opts map
 		if err != nil {
 			return
 		}
+		// if the err == nil && partitionStats == nil, it means we lack the partition-level stats which the physicalID is equal to partitionID.
+		if partitionStats == nil {
+			var errMsg string
+			if isIndex == 0 {
+				errMsg = fmt.Sprintf("`%s`", tableInfo.Name.L)
```
Comment (qw4990) [inline statistics/handle/handle.go:374]: Could we also print its column name here?

Comment (qw4990) [reply statistics/handle/handle.go:374]: Rest LGTM. Please address this, @rebelice .

**Review by qw4990 — APPROVED** [review]

**Pull request merged**